src/myfunctions.js

```diff
@@ -0,0 +1,100 @@
+    editBtn.classList.toggle('remove-icon-active');
+    deleteBtn.classList.toggle('icon2');
+    listText.classList.toggle('listContent-disable');
+    list.classList.toggle('changeBg');
+    const getting = JSON.parse(localStorage.getItem('list'));
+    const empty = [];
+    const selectInput = document.querySelectorAll('.input-div');
+    for (let i = 0; i < getting.length; i += 1) {
+      if (selectInput[i].classList.contains('changeBg')) {
+        getting[i].completed = true;
+      } else {
+        getting[i].completed = false;
+      }
+      empty.push(getting[i]);
+      localStorage.setItem('list', JSON.stringify(empty));
+    }
+  });
+  const clearAll = document.querySelector('#clear');
+  clearAll.addEventListener('click', () => {
+    const getting = JSON.parse(localStorage.getItem('list'));
+    const variable = document.querySelectorAll('.changeBg');
+    for (let i = 0; i < variable.length; i += 1) {
+      form.removeChild(variable[i]);
+    }
+    const empty = [];
+    for (let i = 0; i < getting.length; i += 1) {
+      if (getting[i].completed === true) {
+        continue; // eslint-disable-line
+      }
+      empty.push(getting[i]);
+    }
+    localStorage.setItem('list', JSON.stringify(empty));
+  });
+
+  deleteBtn.addEventListener('click', () => {
+    form.removeChild(list);
+    const getFromLocalStorage = JSON.parse(localStorage.getItem('list'));
+    const result = getFromLocalStorage.filter((word) => word.description === listText.textContent);
+    const empty = [];
+    for (let i = 0; i < getFromLocalStorage.length; i += 1) {
+      if (result[0].description === getFromLocalStorage[i].description) {
+        continue; // eslint-disable-line
+      }
+      empty.push(getFromLocalStorage[i]);
+    }
+    localStorage.setItem('list', JSON.stringify(empty));
+  });
+
+  editBtn.addEventListener('click', () => {
+    const editInput = document.createElement('input');
+    editInput.type = 'text';
+    editInput.className = 'listContent';
+    editInput.style.backgroundColor = '#fffed3';
+    list.style.backgroundColor = '#fffed3';
+    editInput.value = listText.textContent;
+    list.replaceChild(editInput, listText);
+    editInput.addEventListener('keypress', (e) => {
+      if (e.key === 'Enter' && editInput.value) {
+        const getting = JSON.parse(localStorage.getItem('list'));
+        const result = getting.filter((word) => word.description === listText.textContent);
+        const empty = [];
+        for (let i = 0; i < getting.length; i += 1) {
+          if (getting[i].index === result[0].index) {
+            getting[i].description = editInput.value;
+          }
+          empty.push(getting[i]);
+          localStorage.setItem('list', JSON.stringify(empty));
+        }
+        list.replaceChild(listText, editInput);
+        listText.textContent = editInput.value;
+        list.style.backgroundColor = '#fff';
+      }
+    });
+  });
+};
+
+const addToLocalStorage = () => {
```

Comment:
- [ ] All changes to the To-Do List should be saved in local storage, as of now, it doesn't save the correct data (it saves undefined to the `list` key)

![Screenshot_20220610_173144](https://user-images.githubusercontent.com/25227022/173111695-a51aa911-3fd2-4dd1-95ea-aff215a52e9d.png)